src/services/swap.tsx

```diff
@@ -62,43 +62,46 @@ const makeSwapOffer = ({
   };
 
   console.info('OFFER CONFIG: ', offerConfig);
-  return E(walletP).addOffer(offerConfig);
+  wallet.addOffer(offerConfig);
 };
 
 export const doSwap = async (context: SwapContext) => {
   const {
-    setToastId,
-    setSwapped,
-    setCurrentOfferId,
     addError,
-    swapped,
     toPurse,
     fromPurse,
     fromValue,
     toValue,
+    setSwapped,
+    walletUiUrl,
   } = context;
 
-  if (swapped) {
-    addError(SwapError.IN_PROGRESS);
-    return;
-  } else if (!(fromPurse && toPurse)) {
+  if (!(fromPurse && toPurse)) {
     addError(SwapError.NO_BRANDS);
     return;
   } else if (!(toValue && toValue > 0n && fromValue && fromValue > 0n)) {
     addError(SwapError.EMPTY_AMOUNTS);
     return;
   }
 
-  const offerId = await makeSwapOffer(context);
-  setCurrentOfferId(offerId);
+  await makeSwapOffer(context);
   setSwapped(true);
-  setToastId(
-    toast('Please approve the offer in your wallet.', {
-      ...defaultToastProperties,
-      type: toast.TYPE.INFO,
-      progress: undefined,
-      hideProgressBar: true,
-      autoClose: false,
-    })
+  setTimeout(() => {
+    setSwapped(false);
+  }, 3000);
+  toast.success(
```

Comment:
very odd for a _service_ to be JSX. Looks like this is the only reason to make this service .tsx. Instead please move this component to another file and have this call toast or component create from there

Reply:
Yea I think this was a clue that some concerns needed to be separated. Just changed this service to be more decoupled from UI state